### PR TITLE
policies/expression: add support for dynamic variables

### PR DESCRIPTION
**Pull request by rissson**
## Details

Variables that can be loaded from files and be made available to expression policies. Useful for getting secrets from the filesystem.

Note to reviewers: wanted to have a bit of fun coding a small thing, and a friend said that it was annoying to copy stuff between several expression policies so I made this. Treat it as a community contribution.

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [x] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [x] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [x] The code has been formatted (`make web`)
-   [ ] The translation files have been updated (`make i18n-extract`)

If applicable

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make website`)

### CODE REVIEW

File: website/docs/policies/expression.mdx [hunk 1]
```diff
@@ -16,6 +16,39 @@ return False
...
+
+Creating a variable named `my_var` with a value `this is the value` and associating this variable to the expression will result in the following `ak_variables` dictionary being available to use in the expression:
+
+```python
+ak_variables["my_var"] # This is equal to `this_is_the_value`
+```
+
+### External variables
+
+To use externally managed variables (for instance, to pass a secret as a file to authentik), you can use the discovery feature.
```
Comment (tanberry) [inline website/docs/policies/expression.mdx:31]: Can you explain a bit more what you mean by "the discovery feature"?

File: website/docs/policies/expression.mdx [hunk 2]
```diff
@@ -16,6 +16,39 @@ return False
...
+
+The docker-compose installation maps `data/variables` directory to `/data/variables`, you can simply use this directory to store your variables.
```
Comment (tanberry) [inline website/docs/policies/expression.mdx:33]:
```suggestion
The Docker Compose installation maps the `data/variables` directory to the `/data/variables`; you can simply use this directory to store your variables.
```

File: website/docs/policies/expression.mdx [hunk 3]
```diff
@@ -16,6 +16,39 @@ return False
...
+
+For Kubernetes, you can map custom configmaps/secrets/volumes under /data/variables.
```
Comment (tanberry) [inline website/docs/policies/expression.mdx:35]:
```suggestion
For Kubernetes, you can map custom `configmaps/secrets/volumes` under `/data/variables`.
```

File: website/docs/policies/expression.mdx [hunk 4]
```diff
@@ -16,6 +16,39 @@ return False
...
+The name of the variable will be the full path from the `/data/variables/` directory. For instance:
+
+```
+data/variables/
+├── baz
+│   └── bar.baz  # The variable will be named `baz/bar.baz`
+└── foo.bar  # The variable will be named `foo.bar`
+```
+
+Note that file contents are not stripped, and may contain an extra `\n` at the end.
```
Comment (tanberry) [inline website/docs/policies/expression.mdx:48]:
```suggestion
Note that file contents are not stripped, and might contain an extra `\n` at the end.
```

**Review by tanberry — APPROVED** [review]
A few nitpicks, but looks good and sounds fun!

**Comment by BeryJu** [comment]
Closing in favour of https://docs.goauthentik.io/docs/policies/expression#ak_call_policyname-str-kwargs---policyresult

**Pull request closed without merging**